cli/dcoscli/marathon/main.py

```diff
@@ -364,22 +364,27 @@ def _add(app_resource):
     :returns: process return code
     :rtype: int
     """
-    application_resource = _get_resource(app_resource)
+    application_resources = _get_resource(app_resource)
 
     # Add application to marathon
     client = marathon.create_client()
 
     # Check that the application doesn't exist
-    app_id = client.normalize_app_id(application_resource['id'])
+    if not isinstance(application_resources, list):
+        application_resources = [application_resources]
 
-    try:
-        client.get_app(app_id)
-    except DCOSException as e:
-        logger.exception(e)
-    else:
-        raise DCOSException("Application '{}' already exists".format(app_id))
+    for application_resource in application_resources:
+        app_id = client.normalize_app_id(application_resource['id'])
+
+        try:
+            client.get_app(app_id)
+        except DCOSException as e:
+            logger.exception(e)
+        else:
+            raise DCOSException("Application '{}' already exists".format(
```

Comment:
If I have a multi app definition esp. as part of a package I would either want all or none of them to be deployed. Half finished deployments are nasty. Likewise if half of the app is already deployed but the other half is missing there's clearly something wrong. In which case I'd either want to remove the existing apps and do a clean deployment or update the group (as soon as we have group support).